modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -230,8 +230,10 @@ enum VideoWriterProperties {
   VIDEOWRITER_PROP_HW_DEVICE       = 7, //!< (**open-only**) Hardware device index (select GPU if multiple available). Device enumeration is acceleration type specific.
   VIDEOWRITER_PROP_HW_ACCELERATION_USE_OPENCL= 8, //!< (**open-only**) If non-zero, create new OpenCL context and bind it to current thread. The OpenCL context created with Video Acceleration context attached it (if not attached yet) for optimized GPU data copy between cv::UMat and HW accelerated encoder.
   VIDEOWRITER_PROP_RAW_VIDEO = 9, //!< (**open-only**) Set to non-zero to enable encapsulation of an encoded raw video stream. Each raw encoded video frame should be passed to VideoWriter::write() as single row or column of a \ref CV_8UC1 Mat. \note If the key frame interval is not 1 then it must be manually specified by the user. This can either be performed during initialization passing \ref VIDEOWRITER_PROP_KEY_INTERVAL as one of the extra encoder params  to \ref VideoWriter::VideoWriter(const String &, int, double, const Size &, const std::vector< int > &params) or afterwards by setting the \ref VIDEOWRITER_PROP_KEY_FLAG with \ref VideoWriter::set() before writing each frame. FFMpeg backend only.
-  VIDEOWRITER_PROP_KEY_INTERVAL = 10, //!< (**open-only**) Set the key frame interval using raw video encapsulation (\ref VIDEOWRITER_PROP_RAW_VIDEO != 0). Defaults to 1 when not set. FFMpeg backend only.
-  VIDEOWRITER_PROP_KEY_FLAG = 11, //!< Set to non-zero to signal that the following frames are key frames or zero if not, when encapsulating raw video (\ref VIDEOWRITER_PROP_RAW_VIDEO != 0). FFMpeg backend only.
+  VIDEOWRITER_PROP_KEY_INTERVAL = 10, //!< (**open-only**) Set the key frame interval using raw video encapsulation (\ref VIDEOWRITER_PROP_RAW_VIDEO != 0). Defaults to 1 when not set. FFmpeg back-end only.
+  VIDEOWRITER_PROP_KEY_FLAG = 11, //!< Set to non-zero to signal that the following frames are key frames or zero if not, when encapsulating raw video (\ref VIDEOWRITER_PROP_RAW_VIDEO != 0). FFmpeg back-end only.
+  VIDEOWRITER_PROP_PTS_INDEX = 12, //!< Specifies the frame presentation index for each frame when encapsulating raw video (\ref VIDEOWRITER_PROP_RAW_VIDEO != 0). This flag is necessary when encapsulating externally encoded video where the decoding order differs from the presentation order, such as in GOP patterns with bi-directional B-frames.  FFmpeg back-end only.
+  VIDEOWRITER_PROP_B_FRAME_PRESENTATION_DELAY = 13, //!< Maximum delay (in frames) between a B-frame's decoding and presentation. For example, in a GOP with presentation order IBP and decoding order IPB, this value would be 1, as the B-frame is decoded third but presented second.
```

Comment:
It'll be great to add some references to `cv::VideoCapture` properties. It's not obvious, if the properties are needed and how to extract them from the encoded stream.

Reply:
I mean slightly different thing. The VideoWriter user should get values for the new properties somehow. I see 2 sources:
- get something for cv::VideoCapture and promote it to writer (good case). For the case we need a 
- dig in IP camera settings or encoder settings manually (bad case). The case means that we cannot implement generic code that works for all streams.

Reply:
Right I was thinking of streams provided by external encoders (`cudacodec::VideoWriter`) not of the encapsulation of RTSP streams.

For the first parameter (`VIDEOWRITER_PROP_PTS_INDEX`), if the frame rate is provided in the rtsp stream it is equivalent to dividing the current position of the video file (`CAP_PROP_POS_MSEC`) by the duration of the frame (i.e. 1000.0/`VideoCapture::get(CAP_PROP_FPS)` if the frame rate is fixed.  If the frame rate isn't fixed then we would need to create an extra property inside `cv::VideoCapture` to provide this information?

I am not sure if the second parameter (`VIDEOWRITER_PROP_B_FRAME_PRESENTATION_DELAY`) is available inside FFmpeg but if it is (for all versions) we would need to create this extra property inside `cv::VideoCapture` as well.

Reply:
@asmorkalov I've updated VideoCapture to expose the required properties.